tests/host.rs

```diff
@@ -0,0 +1,150 @@
+#![cfg(feature = "providers")]
+
+use core::str;
+use core::time::Duration;
+
+use std::net::Ipv4Addr;
+
+use anyhow::{anyhow, Context};
+use tokio::time::sleep;
+use tokio::try_join;
+use tracing_subscriber::prelude::*;
+use wasmcloud_test_util::lattice::config::assert_config_put;
+use wasmcloud_test_util::provider::{assert_start_provider, StartProviderArgs};
+use wasmcloud_test_util::{
+    component::assert_scale_component, host::WasmCloudTestHost,
+    lattice::link::assert_advertise_link,
+};
+
+use test_components::RUST_HTTP_HELLO_WORLD;
+
+pub mod common;
+use common::free_port;
+use common::nats::start_nats;
+use common::providers;
+
+const LATTICES: &[&str] = &["default", "test-lattice"];
+const COMPONENT_ID: &str = "http_hello_world";
+
+#[tokio::test]
+async fn host_multiple_lattices() -> anyhow::Result<()> {
+    tracing_subscriber::registry()
```

Comment:
In the interest of sanity, I would love to see some more commands tested here for the host like labels being put/deleted, handle retrieving links,claims, and maybe even a subscription on `wasmbus.evt.*.host_heartbeat` to confirm that you get a heartbeat for both lattices.

I think it would also be nice to have some failure cases captured here, or potentially in a different test. Are there any opportunities that you can see that may cause unexpected behavior, for example:
1. Make sure a host doesn't respond to a message for a lattice it doesn't belong to (I think all subjects include the lattice, so this might just be not a useful test)
2. What if I try and make my lattice name `*.>`?
3. What if I create config in lattice `default` and then start a component in `test-lattice` that uses that config, that should fail
4. ...?

What I'm really looking for, with the large amount of changes here, is to be sure that existing functionality is preserved and that the multiple lattices feature doesn't allow for over-reaching that boundary. I don't see anything that would cause that, but if we capture some of the valid and invalid cases here we can move forward with that confidence

Reply:
I added a number of tests which cover most everything you pointed out, along with a scale test that shows we can run at least 2000 lattices on a host at the same time (admittedly without any workloads).

Re: `*.>`, that actually would have failed outside of any of my changes because the lattice data buckets would fail to create with any of those characters. I added an explicit check for them so that we get a sensible error though.